src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -3691,6 +3691,108 @@ .maxstack  4
 ");
         }
 
+        [Fact]
+        public void InferredArrayInitializer()
```

Comment:
>InferredArrayInitializer [](start = 20, length = 24)

I think we should also test when:
- there is a difference in return types
- inference fails
- the only difference is calling convention
- etc.

Reply:
This is marked as resolved, but it doesn't look like any tests were added for:
- inference fails
- the only difference is calling convention


---
In reply to: [406862319](https://github.com/dotnet/roslyn/pull/43041#discussion_r406862319) [](ancestors = 406862319)

Reply:
See `NoBestTypeArrayInitializer` in `FunctionPointerTests.cs`

---
In reply to: [406969760](https://github.com/dotnet/roslyn/pull/43041#discussion_r406969760) [](ancestors = 406969760,406862319)

Reply:
> See NoBestTypeArrayInitializer in FunctionPointerTests.cs

I am not sure why different scenarios for the same thing are tested in different files. I personally don't see any advantage in splitting feature testing across different assemblies, unless there is a dependency on specific helpers/infrastructure

---
In reply to: [406970490](https://github.com/dotnet/roslyn/pull/43041#discussion_r406970490) [](ancestors = 406970490,406969760,406862319)